### PR TITLE
Add plugin-based mock mode

### PR DESCRIPTION
**Pull request by andrei-at-luther**

### CODE REVIEW

File: shiroclient/shiroclient.go [hunk 1]
```diff
@@ -194,6 +198,54 @@ func WithDisableWritePolling(disable bool) Config {
...
+	ro, err := rpc.applyConfigs(configs...)
+	if err != nil {
+		return nil, err
+	}
+	return ro.timestampGenerator, nil
+}
+
+// ProbeTimestampGenerator returns the option values required for a
+// call implied by an array of options.
+func ProbeForCall(configs []Config) (context.Context, string, func(context.Context) string, logrus.Fields, string, string, interface{}, map[string][]byte, error) {
```
Comment (sam-at-luther) [inline shiroclient/shiroclient.go:229]: minor: this comment name seems wrong. go lint might catch these.

File: shiroclient/shiroclient.go [hunk 2]
```diff
@@ -373,6 +442,16 @@ func (t *transaction) Reason() string {
 	return t.reason
 }
 
+// Event implements Transaction
+func (t *transaction) Event() []byte {
+	return t.event
+}
+
+// Event implements Transaction
```
Comment (sam-at-luther) [inline shiroclient/shiroclient.go:450]: this comment here doesn't match. (see below comment)

File: shiroclient/shiroclient.go [hunk 3]
```diff
@@ -381,9 +460,6 @@ const (
 	// MethodShiroPhylum is used to call the ShiroPhylum method which returns
 	// an identifier for the current deployed phylum.
 	MethodShiroPhylum = "ShiroPhylum"
-	// MethodUpgrade is used to call the Upgrade method which upgrades the
-	// substrate .
-	MethodUpgrade = "Upgrade"
```
Comment (sam-at-luther) [inline shiroclient/shiroclient.go:386]: do we not use this anymore?

Comment (andrei-at-luther) [reply shiroclient/shiroclient.go:386]: I found it removed in the latest shiroclient, so I ported the deletion into shiroclient-sdk-go

**Review by sam-at-luther — APPROVED** [review]

**Pull request merged**